lib/spack/llnl/util/filesystem.py

```diff
@@ -832,6 +829,12 @@ def escaped_path(path):
                 set_install_permissions(d)
                 copy_mode(s, d)
 
+    for target, d, s in links:
```

Comment:
`d` doesn't exist yet in the above call to `set_install_permissions(d)` (because the `symlink()` was moved from there to here), so I do in fact think that needs to be removed. Does that make sense?

Reply:
@scheibelp 

> does anything set _permissions=True?

Yes, `install_tree()` always sets it to `True`.

> Does that make sense?

In this case, `d` is the destination file or directory. If it's a link and `symlinks=True`, then it's added to the `links` list and we `continue`. If it's a regular file or directory or `symlinks=False`, those are made/copied on the spot. So it seems to me that `d` will exist every time the `set_install_permissions` method is called. 

There is a separate call to `set_install_permissions` in the `for` loop that makes symlinks.

Reply:
Thanks: I missed the `continue`, can you change that to add another tab to the above `if _permissions:` (such that it lives inside the else) - at which point I think the `continue` is no longer needed.

Reply:
Yes I'm definitely wrong about that: I'm trying to figure out a way to do this without `continue` though since I find it hard to read through the logic: maybe something like

```
set_permissions = True
if is_link
    if symlinks:
        # we don't create symlinks as part of this loop: do it later
        set_permissions = False
```

if you disagree feel free to leave it as is.

Reply:
There isn't a very elegant way to write this without the `continue` so I'm inclined to leave it.